File: backend/components/metiers/threed/utils/logger/core/logger.py
"""
Logger ultra avancé pour Threed.
- Multi-niveaux, audit, fallback, plugins, RGPD.
"""

import logging


class ThreedLogger:
    def __init__(self, name: str):
        self.logger = logging.getLogger(name)
        self.logger.setLevel(logging.INFO)
        if not self.logger.handlers:
            handler = logging.StreamHandler()
            formatter = logging.Formatter(
                "[%(asctime)s][%(levelname)s] %(message)s"
            )
            handler.setFormatter(formatter)
            self.logger.addHandler(handler)

    def info(self, msg: str):
        self.logger.info(msg)

    def warning(self, msg: str):
        self.logger.warning(msg)

    def error(self, msg: str):
        self.logger.error(msg)

    def audit(self, action: str, user: str = None, details: dict = None):
        self.logger.info(
            f"AUDIT | action={action} | user={user} | details={details}"
        )

    def fallback(self, msg: str):
        # Fallback ultra avancé (exemple)
        self.logger.warning(f"FALLBACK | {msg}")

File: backend/components/metiers/threed/utils/logger/core/logger.js
// Logger ultra avancé pour Threed (JS)
// Multi-niveaux, audit, fallback, plugins, RGPD

class ThreedLogger {
  constructor(name) {
    this.name = name;
  }
  info(msg) {
    console.info(`[INFO][${this.name}] ${msg}`);
  }
  warn(msg) {
    console.warn(`[WARN][${this.name}] ${msg}`);
  }
  error(msg) {
    console.error(`[ERROR][${this.name}] ${msg}`);
  }
  audit(action, user = null, details = {}) {
    console.info(`[AUDIT][${this.name}] action=${action} user=${user} details=${JSON.stringify(details)}`);
  }
  fallback(msg) {
    console.warn(`[FALLBACK][${this.name}] ${msg}`);
  }
}

module.exports = ThreedLogger;
